subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,269 @@
+    please see http://www.xiph.org/ for how to
+    obtain a copy.
+  ''',
+)
+
+vorbis_dep = dependency('vorbis', version: '>= 1.0.1', required: false)
+
+sdl_dep = dependency('sdl', required: false)
+
+if not sdl_dep.found()
+  message('*** Unable to find SDL -- Not compiling example players ***')
+endif
+
+if (
+  cc.has_header('sys/soundcard.h')
+  or cc.has_header('soundcard.h')
+  or cc.has_header('machine/soundcard.h')
+)
+  if host_machine.system() == 'openbsd'
+    oss_dep = dependency('ossaudio', required: false, disabler: true)
+  else
+    oss_dep = declare_dependency()
+  endif
+else
+  oss_dep = disabler()
+endif
+
+if not oss_dep.found()
+  message('OSS audio support not found -- not compiling player_example')
+endif
+
+libpng_dep = dependency('libpng', required: false)
+
+tiff_dep = dependency('tiff', required: false, disabler: true)
+
+if (
+  tiff_dep.found()
+  and not cc.has_header_symbol(
+    'tiffio.h',
+    'TIFFReadRGBAImage',
+    dependencies: tiff_dep,
+  )
+)
+  tiff_dep = disabler()
+endif
+
+telemetry = get_option('telemetry')
+
+cairo_dep = dependency('cairo', required: false)
+
+if telemetry.allowed()
+  if cairo_dep.found()
+    conf.set('HAVE_CAIRO', true)
+  else
+    message('libcairo not found -- not compiling telemetry output support ')
+  endif
+endif
+
+mem_constraint = get_option('mem-constraint')
+
+if mem_constraint.enabled()
+  conf.set('HAVE_MEMORY_CONSTRAINT', language: 'c')
+endif
+
+asm = get_option('asm').allowed()
+if asm
+  if host_machine.cpu_family() == 'x86_64'
+    conf.set('CPU_x86_64', true)
+    if cc.get_id() != 'msvc'
+      conf.set('OC_X86_ASM', true)
+      conf.set('OC_X86_64_ASM', true)
+    endif
+  elif host_machine.cpu_family() == 'x86'
+    conf.set('OC_X86_ASM', true)
+  endif
+endif
+
+config_h = configure_file(configuration: conf, output: 'config.h')
+
+config_dep = declare_dependency(sources: config_h)
+
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+pkg = import('pkgconfig')
+windows = import('windows')
+
+subdir('include')
+
+subdir('lib')
+
+if get_option('examples').enabled()
+  subdir('examples')
+endif
+
+subdir('tests')
+
+theora_dep = declare_dependency(
+  link_with: libtheora,
+  include_directories: incdir,
+)
+
+theoradec_dep = declare_dependency(
+  link_with: libtheoradec,
+  include_directories: incdir,
+)
+
+theoraenc_dep = declare_dependency(
+  link_with: libtheoradec,
+  include_directories: incdir,
+)
+
+if meson.version().version_compare('>=0.54.0')
+  meson.override_dependency('theora', theora_dep)
+  meson.override_dependency('theoradec', theoradec_dep)
+  meson.override_dependency('theoraenc', theoraenc_dep)
+endif
+
+doc = get_option('doc')
+
+if doc.allowed()
+  doxygen_exe = find_program(
+    'doxygen',
+    required: doc.enabled(),
+    disabler: true,
+  )
+  if not doxygen_exe.found()
+    message('*** doxygen not found, API documentation will not be built')
+  endif
+  doc.disable_auto_if(not doxygen_exe.found())
+endif
```

Comment:
Dunno what all this is. When `-Ddoc=disabled`, this evaluates as `required: false`, doxygen_exe is searched for nevertheless, and then you cast doc from auto to disabled, returning a new immutable object that you then garbage collect.

Reply:
`--enable-doc` in ./configure, here it'd be `-Ddoc=enabled`.